utils/config_file.cc

```diff
@@ -304,6 +315,9 @@ void utils::config_file::read_from_yaml(const char* yaml, error_handler h) {
             continue;
         }
         switch (cfg.status()) {
+        case value_status::Deprecated:
+            h(label, "Option is deprecated", cfg.status());
+            continue;
```

Comment:
@ptrsmrn 

This introduces a bug IMO.
Previously, those options were warned about, but still applied.
Now, this `continue`, rather than `break` stop processing short, before calling `cfg.set_value(node.second);` below.
So what's the difference between `value_status::Deprecated` and `value_status::Invalid`?

Normally, when an interface is deprecated, it continues to function normally, despite emitting a warning, so to give time for the users to replace its usage with something else (that should be printed as well as part of the deprecation warning).  Only when the server stops supporting it, it should be changed to `Invalid`, so it can't be used anymore.